src/ray/core_worker/task_manager.cc

```diff
@@ -52,6 +51,24 @@ void TaskManager::AddPendingTask(const TaskID &caller_id,
   }
 }
 
+void TaskManager::AddPendingTask(const TaskID &caller_id,
+                                 const rpc::Address &caller_address,
+                                 const TaskSpecification &spec, int max_retries) {
+  RAY_CHECK(!spec.IsActorTask());
+  absl::MutexLock lock(&mu_);
+  AddPendingTaskImpl(caller_id, caller_address, spec, max_retries);
+}
+
+void TaskManager::AddPendingActorTask(const TaskID &caller_id,
+                                 const rpc::Address &caller_address,
+                                 const TaskSpecification &spec) {
+  RAY_CHECK(spec.IsActorTask());
+  absl::MutexLock lock(&mu_);
+  AddPendingTaskImpl(caller_id, caller_address, spec, /*max_retries=*/0);
+  pending_actor_tasks_[spec.ActorId()].push_back(spec.TaskId());
+  actor_send_task_callback_(spec);
```

Comment:
Need to be careful about calling this while holding `mu_` (both for possible deadlock and performance)